### PR TITLE
Update rules_haskell - now generates Haskell version macros

### PR DESCRIPTION
**Pull request by aherrmann-da**
- rules_haskell now generates a version macros header file that is passed on to preprocessors such as c2hs or hsc2hs.
- The haskell_import rule was also renamed to haskell_toolchain_library.
- Drops the version macros patch to `unix-compat` which is now redundant.
- Resolves the version macros point in https://github.com/digital-asset/daml/issues/582.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/docs/source/support/release-notes.rst), if appropriate

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: pipeline/samples/bazel/haskell/BUILD [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
     "@io_tweag_rules_haskell//haskell:haskell.bzl",
     "haskell_binary",
-    "haskell_import",
     "haskell_library",
+    "haskell_toolchain_library",
 )
 load("@ai_formation_hazel//:hazel.bzl", "hazel_library")
 
-haskell_import(name = "base")
+haskell_toolchain_library(name = "base")
```
Comment (cocreature) [inline pipeline/samples/bazel/haskell/BUILD]: We should just delete the samples at this point but that is probably better done in a separate PR.

Comment (cocreature) [reply pipeline/samples/bazel/haskell/BUILD]: Oh sorry, I didn’t realize this wasn’t merged yet when I made the PR to remove the samples. But it should hopefully be easy to fix.

Comment (aherrmann-da) [reply pipeline/samples/bazel/haskell/BUILD]: No worries, rebase was just a `git rm`.
